zkevm-circuits/src/tx_circuit.rs

```diff
@@ -0,0 +1,516 @@
+//! The transaction circuit implementation.
+
+// Naming notes:
+// - *_be: Big-Endian bytes
+// - *_le: Little-Endian bytes
+
+mod sign_verify;
+
+use crate::util::{random_linear_combine_word as rlc, Expr};
+use eth_types::{
+    geth_types::Transaction, Address, Field, ToBigEndian, ToLittleEndian, ToScalar, Word,
+};
+use ff::PrimeField;
+use group::GroupEncoding;
+use halo2_proofs::{
+    circuit::{AssignedCell, Layouter, Region, SimpleFloorPlanner},
+    plonk::{Advice, Circuit, Column, ConstraintSystem, Error},
+    poly::Rotation,
+};
+use itertools::Itertools;
+use lazy_static::lazy_static;
+use libsecp256k1;
+use log::error;
+use num::Integer;
+use num_bigint::BigUint;
+use rlp::RlpStream;
+use secp256k1::Secp256k1Affine;
+use sha3::{Digest, Keccak256};
+use sign_verify::{SignData, SignVerifyChip, SignVerifyConfig};
+pub use sign_verify::{POW_RAND_SIZE, VERIF_HEIGHT};
+use std::convert::TryInto;
+use std::marker::PhantomData;
+use subtle::CtOption;
+
+lazy_static! {
+    // Scalar
+    static ref SECP256K1_Q: BigUint = BigUint::from_slice(&[
+        0xd0364141, 0xbfd25e8c,
+        0xaf48a03b, 0xbaaedce6,
+        0xfffffffe, 0xffffffff,
+        0xffffffff, 0xffffffff,
+    ]);
+}
+
+fn recover_pk(v: u8, r: &Word, s: &Word, msg_hash: &[u8; 32]) -> Result<Secp256k1Affine, Error> {
+    let r_be = r.to_be_bytes();
+    let s_be = s.to_be_bytes();
+    let mut r = libsecp256k1::curve::Scalar::from_int(0);
+    let r_overflow: bool = r.set_b32(&r_be).into();
+    let mut s = libsecp256k1::curve::Scalar::from_int(0);
+    let s_overflow: bool = s.set_b32(&s_be).into();
+    if r_overflow || s_overflow {
```

Comment:
Have you tried to use  `libsecp256l1::Signature::parse_standard` ?

Reply:
Thanks for this suggestion! This simplifies the code a bit :) Done in https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/484/commits/e3a4615f7fdf2961909ae0eb66491236ed57a7d1